PhysicsTools/PatAlgos/plugins/PATMuonMerger.cc

```diff
@@ -33,24 +38,38 @@ class PatMuonMerger : public edm::stream::EDProducer<> {
   edm::EDGetTokenT<std::vector<pat::Muon>> muonToken_;
   edm::EDGetTokenT<std::vector<pat::PackedCandidate>> pfCandToken_;
   edm::EDGetTokenT<std::vector<pat::PackedCandidate>> lostTrackToken_;
+  
+
 };
 
-PatMuonMerger::PatMuonMerger(const edm::ParameterSet& iConfig)
+PATMuonMerger::PATMuonMerger(const edm::ParameterSet& iConfig)
     : muons_(iConfig.getParameter<edm::InputTag>("muons")),
-      muonsCut_(iConfig.existsAs<std::string>("muonCut") ? iConfig.getParameter<std::string>("muonCut") : ""),
+      muonsCut_(iConfig.getParameter<std::string>("muonCut")),
       pfCandidate_(iConfig.getParameter<edm::InputTag>("pfCandidates")),
-      pfCandidateCut_(
-          iConfig.existsAs<std::string>("pfCandidatesCut") ? iConfig.getParameter<std::string>("pfCandidatesCut") : ""),
+      pfCandidateCut_(iConfig.getParameter<std::string>("pfCandidatesCut")),
       lostTrack_(iConfig.getParameter<edm::InputTag>("otherTracks")),
-      lostTrackCut_(iConfig.existsAs<std::string>("lostTrackCut") ? iConfig.getParameter<std::string>("lostTrackCut")
-                                                                  : "") {
+      lostTrackCut_(iConfig.getParameter<std::string>("lostTrackCut")) {
   muonToken_ = consumes<std::vector<pat::Muon>>(muons_);
   pfCandToken_ = consumes<std::vector<pat::PackedCandidate>>(pfCandidate_);
   lostTrackToken_ = consumes<std::vector<pat::PackedCandidate>>(lostTrack_);
   produces<std::vector<pat::Muon>>();
 }
 
-void PatMuonMerger::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+
+void
+PATMuonMerger::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // mergedMuons
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("muonCut", "pt>15 && abs(eta)<2.4");
+  desc.add<edm::InputTag>("otherTracks", edm::InputTag("lostTracks"));
+  desc.add<edm::InputTag>("pfCandidates", edm::InputTag("packedPFCandidates"));
+  desc.add<std::string>("pfCandidatesCut", "pt>15 && abs(eta)<2.4");
+  desc.add<edm::InputTag>("muons", edm::InputTag("slimmedMuons"));
+  desc.add<std::string>("lostTrackCut", "pt>15 && abs(eta)<2.4");
+  descriptions.add("mergedMuons", desc);
```

Comment:
Do not give the same name to the automatically created module, and the one  explicitly defined in ```patMuonMerger_cfi.py```
Here it could be "mergedMuonsDef" or "mergedMuonsNoCuts", or whatever else you prefer